src/wp-includes/rest-api/endpoints/class-wp-rest-plugins-controller.php

```diff
@@ -51,10 +51,16 @@ public function register_routes() {
 					'args'                => array(
 						'slug'   => array(
 							'type'        => 'string',
-							'required'    => true,
+							'required'    => false,
 							'description' => __( 'WordPress.org plugin directory slug.' ),
 							'pattern'     => '[\w\-]+',
 						),
+						'url'    => array(
+							'type'        => 'string',
+							'required'    => false,
+							'description' => __( 'A URL to a zip or tar archive.' ),
+							'pattern'     => '^https?://[^\/]*/.*(zip|tar|tar\.gz)$',
```

Comment:
Instead of using a `pattern`, this should use the `format` keyword set to `uri`.

Reply:
Can it use both? i.e. use `format` to check is it a URL, and use a pattern to check it's an allowed archive format.

Reply:
Yes, it's additive, so all keywords provided will be evaluated. I'd then remove the URL validation portion, and just focus on it ending with one of the expected extensions.

Reply:
OK. I consider this part resolved: https://www.phpliveregex.com/p/GAT

Reply:
Although I set `'required'    => false,` on the two params. Can I leave that out? I think it's clearer to leave it out, rather than suggesting it fully is required or is not required.

Reply:
Correct, the keyword can be omitted to treat it as a non-required field.

Reply:
Done.